### PR TITLE
support IIGS mouse

### PR DESCRIPTION
**Pull request by peterferrie**

### CODE REVIEW

File: mouse.gs.s [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+GsDisableMouse:
+	pla
+	pla					; discard return address
+	stz WG_MOUSEACTIVE
+	jsr WGUndrawPointer			; Be nice if we're disabled during a program
+	pla
+	rts
+
+;-------------------------------
```
Comment (blondie7575) [inline mouse.gs.s]: Can you rename this file so it doesn't use a period in the name? It's confusing as it looks like two extensions. Some OSes or tools may not like that. An underscore or camel-case would be in keeping with the naming conventions of the project.

File: mouse.s [hunk 2]
```diff
@@ -75,6 +75,10 @@ WGEnableMouse:
 
 	SETSWITCH PAGE2OFF
 
+	sec
+	.byte $C2, $01 ;REP #$01	; clear carry flag on 65816
```
Comment (blondie7575) [inline mouse.s]: We're relying on the fact that the 65C02 treats undefined opcodes as NOPs here, so we should document that obscure fact in the comment. Someone may wish to support the NMOS 6502 someday, and could be burned by this ($C2 is suspected to halt some models of NMOS 6502).

**Review by blondie7575 — CHANGES_REQUESTED** [review]

**Comment by blondie7575** [comment]
Hi Peter,

Looks good! I left a couple of minor comments.

Q

**Comment by peterferrie** [comment]
Done and done.

**Pull request merged**